sdk/go/common/workspace/project.go

```diff
@@ -357,20 +357,29 @@ func ValidateProject(raw interface{}) error {
 
 	var errs *multierror.Error
 	var appendError func(err *jsonschema.ValidationError)
+	locToErr := make(map[string]*strings.Builder)
+	errorf := func(path, message string, args ...interface{}) error {
+		contract.Require(path != "", "path")
+		return fmt.Errorf("%s: %s", path, fmt.Sprintf(message, args...))
+	}
 	appendError = func(err *jsonschema.ValidationError) {
-		if err.InstanceLocation != "" && err.Message != "" {
-			errorf := func(path, message string, args ...interface{}) error {
-				contract.Require(path != "", "path")
-				return fmt.Errorf("%s: %s", path, fmt.Sprintf(message, args...))
+		if err.InstanceLocation != "" && err.Message != "" && len(err.Causes) == 0 {
+			if e, ok := locToErr[err.InstanceLocation]; ok {
```

Comment:
Concatenating errors from the same location cleans it up a bit more
```
2 errors occurred:
        * #/config/bucketNamePrefix: expected string, but got object; expected integer, but got object; expected boolean, but got object; expected array, but got object
        * #/config/bucketNamePrefix/type: value must be one of "string", "integer", "boolean", "array"
        ```

Reply:
This is a major improvement! Would it be possible to go one step further with the first error:
```
2 errors occurred:
        * #/config/bucketNamePrefix: expected a string, integer, boolean or array, but got object
        * #/config/bucketNamePrefix/type: value must be one of "string", "integer", "boolean", "array"
```

I imagine we can do a post-folding step where we combine all messages of the form `expected a \([a-z]+\), but got \([a-z]+\)`.